flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -231,6 +231,28 @@ ANY, and(logical(LogicalTypeRoot.BOOLEAN), LITERAL)
                             "org.apache.flink.table.runtime.functions.scalar.ArrayContainsFunction")
                     .build();
 
+    public static final BuiltInFunctionDefinition ARRAY_SORT =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("ARRAY_SORT")
+                    .kind(SCALAR)
+                    .inputTypeStrategy(
+                            or(
+                                    sequence(ARRAY_FULLY_COMPARABLE),
+                                    sequence(
+                                            ARRAY_FULLY_COMPARABLE,
+                                            InputTypeStrategies.explicit(
+                                                    DataTypes.BOOLEAN().nullable())),
+                                    sequence(
+                                            ARRAY_FULLY_COMPARABLE,
+                                            InputTypeStrategies.explicit(
+                                                    DataTypes.BOOLEAN().nullable()),
+                                            InputTypeStrategies.explicit(
+                                                    DataTypes.BOOLEAN().nullable()))))
```

Comment:
Let's use `logical(LogicalTypeRoot.BOOLEAN)` here.

That way the output strategy have a chance of being `NOT NULL`.
In the current state the output has always `NULLABLE` type, because 2nd and 3rd aguments are nullable. If you do: `logical(LogicalTypeRoot.BOOLEAN)` and pass e.g. literals: `ARRAY(1, 2, 3), true, true`, the output type will be `NOT NULL`, because all three arguments have `NOT NULL` types.

The strategy here does not only perform validation, but it also does type coercion.

Reply:
ok, I will update.